### PR TITLE
Webapp

### PR DESCRIPTION
**Pull request by delhomer**
This PR introduces a brand new way for visualizing deep learning predictions through a Flask web application.
The application is quite simple, as no real focus has been done on formatting web pages. However, it provides a first complete picture of the project results.

In order to make the Flask app happen, some symbolic link must be designed:
+ `deeposlandia/static/shapes` that points on shape image folder (the folder that contains a `images` and a `labels` subrepositories), for demo purpose.
+ `deeposlandia/static/mapillary` that points on mapillary image folder for demo purpose (*see* previous point).
+ `deeposlandia/static/predicted_images` that points to a server-side temporary folder to store predicted images (labelled versions of tested images).
+ `deeposlandia/static/sample_image` that points to `images` repository within the `deeposlandia` project, so as to display example images at web page first visits.

This PR fixes issue #55 .

### CODE REVIEW

File: deeposlandia/static/app.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+var API_URL = PREFIX + '/api';
+
+// Build the URL with a BASE_URL/<city> suffix based from a DOM element with the
+// "city" dataset attribute, i.e. 'data-city'.
+function cityurl(dom_id) {
+  return API_URL + "/" + document.getElementById(dom_id).dataset.city;
+};
+
+// get the date before today in YYYY-MM-DD string format
+function getYesterday() {
```
Comment (garaud) [inline deeposlandia/static/app.js]: Is this function mandatory? Maybe you don't neet it.

File: deeposlandia/static/app.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Jitenshea
+
+// PREFIX should be defined in the settings.js file
+var API_URL = PREFIX + '/api';
+
+// Build the URL with a BASE_URL/<city> suffix based from a DOM element with the
+// "city" dataset attribute, i.e. 'data-city'.
+function cityurl(dom_id) {
```
Comment (garaud) [inline deeposlandia/static/app.js]: Legacy copy/pasting from jitenshea I think. You can remove it.

File: deeposlandia/webapp.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+PREDICT_FOLDER = os.path.join(PROJECT_FOLDER, 'predicted/')
+os.makedirs(UPLOAD_FOLDER, exist_ok=True)
+os.makedirs(PREDICT_FOLDER, exist_ok=True)
+
+daiquiri.setup(level=logging.INFO)
+logger = daiquiri.getLogger("deeposlandia-webapp")
+
+app = Flask(__name__)
+app.config['ERROR_404_HELP'] = False
+app.config['SWAGGER_UI_DOC_EXPANSION'] = 'list'
```
Comment (garaud) [inline deeposlandia/webapp.py]: I think you don't use Swagger, you can remove it.

File: deeposlandia/webapp.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    if request.method == 'POST':
+        # check if the post request has the file part
+        if 'file' not in request.files:
+            print('No file part')
+            return redirect(request.url)
+        file = request.files['file']
+        # if user does not select file, browser also
+        # submit a empty part without filename
+        if file.filename == '':
+            print('No selected file')
```
Comment (garaud) [inline deeposlandia/webapp.py]: remaining print

**Review by garaud — CHANGES_REQUESTED** [review]
Great work! Congrats.

I think that the symbolic link part with the different image folders could be in a specific configuration file, read by the Flask app when it starts.

It's not very clear to me the name of the different routes: `demo_predictor`, `predictor`, `predictor_demo`. You don't know if it's a route which serves a HTML file, or a route which serves some JSON data. And I think you should add some docstring.

File: deeposlandia/static/app.js (hunk 1 above)
Comment (delhomer) [reply deeposlandia/static/app.js]: Yes, indeed. Actually the whole `app.js` is useless for `deeposlandia`. I remove it.

File: deeposlandia/static/app.js (hunk 2 above)
Comment (delhomer) [reply deeposlandia/static/app.js]: You're write, I remove the file from the project. :)

File: deeposlandia/webapp.py (hunk 3 above)
Comment (delhomer) [reply deeposlandia/webapp.py]: Done.

File: deeposlandia/webapp.py (hunk 4 above)
Comment (delhomer) [reply deeposlandia/webapp.py]: Actually this is a stackoverflow-oriented development issue, there... Maybe we can keep the log as the function behavior still have to be clarified. I replace `print` statements with `utils.logger.info`.

**Comment by delhomer** [comment]
I agree regarding the symbolic link point. I will add a dedicated issue to keep it in our mind.
Additionally, I will add a new commit with docstrings for webapp.py methods.

**Review by garaud — APPROVED** [review]

**Pull request merged**